### PR TITLE
Implement a dark theme

### PR DESCRIPTION
**Pull request by David-DB88**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing application instances)

**What changes did you make?** (Give an overview)
Created Dark mode on kafka-ui

**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [ ] No need to
- [x] Manually (please, describe, if necessary)
- [ ] Unit checks
- [ ] Integration checks
- [ ] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by Haarolean — CHANGES_REQUESTED** [review]
@David-DB88 a few things I noticed:
1. Code editors (smart filters, schemas) are still white. Is that possible to change them as well?
2. Inputs borders colors are invalid
image
3. "show offline clusters" label color is invalid
image
4. saved filters names labels text color is invalid
image
5. connectors topic' names labels' text background color is white:
image

**Comment by David-DB88** [comment]
yes i know i just pushed it for now but i will change what you noticed, thank you

File: kafka-ui-react-app/src/components/Schemas/Details/LatestVersion/LatestVersionItem.styled.tsx [hunk 1]
```diff
@@ -1,11 +1,10 @@
 import Heading from 'components/common/heading/Heading.styled';
 import React from 'react';
 import styled from 'styled-components';
-import theme from 'theme/theme';
 
 export const Wrapper = styled.div`
   width: 100%;
-  background-color: ${theme.layout.stuffColor};
+  background-color: ${(props) => props.theme.layout.stuffColor};
```
Comment (Mgrdich) [inline kafka-ui-react-app/src/components/Schemas/Details/LatestVersion/LatestVersionItem.styled.tsx:7]: Let's try to destructure what we can to make to short.

File: kafka-ui-react-app/src/components/common/Icons/ArrowDownIcon.tsx [hunk 2]
```diff
@@ -1,15 +1,20 @@
...
-    xmlns="http://www.w3.org/2000/svg"
-    viewBox="0 0 384 512"
-    width="10"
-    height="10"
-  >
-    {/* Font Awesome Pro 6.1.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license (Commercial License) Copyright 2022 Fonticons, Inc. */}
-    <path d="M374.6 310.6l-160 160C208.4 476.9 200.2 480 192 480s-16.38-3.125-22.62-9.375l-160-160c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0L160 370.8V64c0-17.69 14.33-31.1 31.1-31.1S224 46.31 224 64v306.8l105.4-105.4c12.5-12.5 32.75-12.5 45.25 0S387.1 298.1 374.6 310.6z" />
-  </svg>
-);
+const ArrowDownIcon: React.FC = () => {
```
Comment (Mgrdich) [inline kafka-ui-react-app/src/components/common/Icons/ArrowDownIcon.tsx:4]: Here the arrow icon is being customized with some metrics color that is unrelated to it .

Either we have to pass colour as a prop and put some default color , or make the variable name independent from the metrics.

I would go with the first case.

File: kafka-ui-react-app/src/components/common/Icons/ClockIcon.tsx [hunk 3]
```diff
@@ -1,15 +1,20 @@
...
-    xmlns="http://www.w3.org/2000/svg"
-    viewBox="0 0 512 512"
-    width={10}
-    height={10}
-  >
-    {/* Font Awesome Pro 6.1.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license (Commercial License) Copyright 2022 Fonticons, Inc. */}
-    <path d="M232 120C232 106.7 242.7 96 256 96C269.3 96 280 106.7 280 120V243.2L365.3 300C376.3 307.4 379.3 322.3 371.1 333.3C364.6 344.3 349.7 347.3 338.7 339.1L242.7 275.1C236 271.5 232 264 232 255.1L232 120zM256 0C397.4 0 512 114.6 512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256C0 114.6 114.6 0 256 0zM48 256C48 370.9 141.1 464 256 464C370.9 464 464 370.9 464 256C464 141.1 370.9 48 256 48C141.1 48 48 141.1 48 256z" />
-  </svg>
-);
+const ClockIcon: React.FC = () => {
```
Comment (Mgrdich) [inline kafka-ui-react-app/src/components/common/Icons/ClockIcon.tsx:4]: Same thing as the other icon.

File: kafka-ui-react-app/src/components/common/Icons/FileIcon.tsx [hunk 4]
```diff
@@ -1,15 +1,20 @@
...
-    xmlns="http://www.w3.org/2000/svg"
-    viewBox="0 0 384 512"
-    width="10"
-    height="10"
-  >
-    {/* Font Awesome Pro 6.1.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license (Commercial License) Copyright 2022 Fonticons, Inc. */}
-    <path d="M365.3 93.38l-74.63-74.64C278.6 6.742 262.3 0 245.4 0L64-.0001c-35.35 0-64 28.65-64 64l.0065 384c0 35.34 28.65 64 64 64H320c35.2 0 64-28.8 64-64V138.6C384 121.7 377.3 105.4 365.3 93.38zM336 448c0 8.836-7.164 16-16 16H64.02c-8.838 0-16-7.164-16-16L48 64.13c0-8.836 7.164-16 16-16h160L224 128c0 17.67 14.33 32 32 32h79.1V448zM96 280C96 293.3 106.8 304 120 304h144C277.3 304 288 293.3 288 280S277.3 256 264 256h-144C106.8 256 96 266.8 96 280zM264 352h-144C106.8 352 96 362.8 96 376s10.75 24 24 24h144c13.25 0 24-10.75 24-24S277.3 352 264 352z" />
-  </svg>
-);
+const FileIcon: React.FC = () => {
```
Comment (Mgrdich) [inline kafka-ui-react-app/src/components/common/Icons/FileIcon.tsx:4]: Same here.

**Review by Haarolean — CHANGES_REQUESTED** [review]
image
tabs are transparent

File: kafka-ui-react-app/src/components/App.tsx [hunk 5]
```diff
@@ -39,21 +43,85 @@ const queryClient = new QueryClient({
...
   },
 });
+const options = [
+  {
+    label: (
+      <>
+        <AutoIcon /> Auto theme
+      </>
+    ),
+    value: 'Auto theme',
```
Comment (Mgrdich) [inline kafka-ui-react-app/src/components/App.tsx:53]: since value labels are not visible to the user, lets make them lowercase and with `_` , `auto_theme`

**Review by Haarolean — CHANGES_REQUESTED** [review]
This component design doesn't correspond with figma
image
figma:
image

There's no indication that a link is indeed a link:
image

File: kafka-ui-react-app/src/components/App.tsx [hunk 6]
```diff
@@ -38,16 +38,18 @@ const queryClient = new QueryClient({
 });
 
 const App: React.FC = () => {
+  const [isDarkMode, setDarkMode] = React.useState<boolean>(false);
+
   return (
     <QueryClientProvider client={queryClient}>
       <GlobalSettingsProvider>
-        <ThemeProvider theme={theme}>
+        <ThemeProvider theme={isDarkMode ? darkTheme : theme}>
```
Comment (jindradev) [inline kafka-ui-react-app/src/components/App.tsx:46]: Am I dreaming? true/false for theming rework of this scale? I had hopes for you guys this afternoon. Hope you will iterate past this soon. A simple palette would do, not even CSS variables.

File: kafka-ui-react-app/src/components/common/PropertiesList/PropertiesList.styled.tsx [hunk 7]
```diff
@@ -12,11 +12,11 @@ export const List = styled.div`
...
-  color: ${({ theme }) => theme.list.label.color};
+  color: ${({ theme }) => theme.default.color.normal};
   white-space: nowrap;
 `;
 
-export const SubText = styled.div`
-  color: ${({ theme }) => theme.list.meta.color};
-  font-size: 12px;
-`;
+// export const SubText = styled.div`
```
Comment (Mgrdich) [inline kafka-ui-react-app/src/components/common/PropertiesList/PropertiesList.styled.tsx:19]: if we don't need the component lets delete it.

Comment (David-DB88) [reply kafka-ui-react-app/src/components/common/PropertiesList/PropertiesList.styled.tsx]: done

**Review by Haarolean — CHANGES_REQUESTED** [review]
image
wrong link position, wrong font height
image
the tooltip still doesn't work, the icon is ugly
image
Unfocused field border color is invalid

**Comment by David-DB88** [comment]
@Haarolean  this can be checked

**Comment by iliax** [comment]
@David-DB88 notify test

**Comment by David-DB88** [comment]
@iliax thanks

**Review by armenuikafka — CHANGES_REQUESTED** [review]
@David-DB88 @Haarolean 
-  could be fix the design of connector config with create connector

https://user-images.githubusercontent.com/104780608/220273723-cc8d5c74-df6e-4e02-bf07-1bfc76b29208.mov

- create and cancel buttons should be in same color or not?
topic create

- source icon is not functional in Brokers page

- key and value previews in topic/messages should be with right colors

https://user-images.githubusercontent.com/104780608/220274650-67fc9726-5cdb-4625-8ba6-46cf79158da3.mov

- whats the difference between auto theme and light theme?

**Comment by BulatKha** [comment]
> * create and cancel buttons should be in same color or not?

Create and Cancel buttons should be of the different color and they look ok on the screenshot. However the 'Add custom parameter' button might need to differ in color. Like it is greyscale in Light theme now.

**Comment by David-DB88** [comment]
@BulatKha  can you please clarify what does not meet the requirements, because all colors are supplied in accordance with the design Screen Shot 2023-02-21 at 15 48 51

**Comment by David-DB88** [comment]
@armenuikafka sure we can change color  create connector design like KSQL DB but but it's done by design,
Create and Cancel buttons in add Topic also done by design,
Source icon if you mean Info Icon than that is now work normal i fixed,
key and value previews in topic/messages now looks normal,
the difference between auto theme and light theme is that when you select auto then during the day it changes  automatically from light to dark theme or opposite, you can check it by changes settings on your laptop

**Comment by armenuikafka** [comment]
@David-DB88 

- can't see the account in a right corner 
dark account

- metrics tab is empty (not sure because of dark mode changes)

- @Haarolean could you please confirm that design is true (there are some Inconsistencies for danger zone, toggle and preview links colors, messages loading in case of live mode etc.) 
dark danger zone
dark toggle color
dark messages colors

- what about success messages design? are we keeping this ?

https://user-images.githubusercontent.com/104780608/221497573-55517b82-c9e6-42c1-8d79-da3db5467e0d.mov

- remove icon for saved filters in messages is red (not as in design)
dark saved filter remove

- pop up in case of deleting saved filter could have the other background color I think (you can see for light mode it looks good)

https://user-images.githubusercontent.com/104780608/221499535-0c5e9b83-a80e-4fa6-b859-f570ec2a9045.mov

- stop analysis has the same design as Start analysis (but are different in design)
dark stop analysis

**Comment by Haarolean** [comment]
@armenuikafka 

>can't see the account in a right corner

that's fine, there's no authentication enabled.

>metrics tab is empty (not sure because of dark mode changes)

that's a configuration issue

>@Haarolean could you please confirm that design is true (there are some Inconsistencies for danger zone, toggle and preview links colors, messages loading in case of live mode etc.)

yes, If something doesn't match, the design should prevail. 

>what about success messages design? are we keeping this ?

that's out of the scope of the issue but seems useless. Please raise a new issue.

**Comment by armenuikafka** [comment]
@David-DB88 

- as I see in a design the "restart analytics" button should be chnaged
restart analytics

- in statistics tab, if partition isn't opened, the color should stay the same, only "-" should be blue (please check also the consumer profile)
dark statistics partition

- @Haarolean in a design some more info, do we need add these to system (I remember we talked about it and system is true, design is not up to dated )
consumer profile

**Comment by David-DB88** [comment]
@armenuikafka when partition is open i did like this
Screen Shot 2023-02-28 at 16 11 39

also in general component styles we have examples like this, normal, hover, active states 
Screen Shot 2023-02-28 at 16 13 49

**Comment by David-DB88** [comment]
@armenuikafka made a changes on danger zone, switch button, plus icon, saved filters remove icon, saved filter background color, stop analysis button.

**Comment by armenuikafka** [comment]
@David-DB88 were can I check the fixes? I think here http://pr2996.internal.kafka-ui.provectus.io/ are not available

**Comment by David-DB88** [comment]
may be some problem with deploy

**Comment by David-DB88** [comment]
@armenuikafka here is the link http://pr2996.internal.kafka-ui.provectus.io/

**Comment by David-DB88** [comment]
@armenuikafka I talked to @Haarolean and he said that if there are already other buttons in the white one, then we do it as a separate task, so this two buttons we can do with separate task 
Screen Shot 2023-03-01 at 15 41 15
Screen Shot 2023-03-01 at 15 41 29

**Comment by armenuikafka** [comment]
> @armenuikafka I talked to @Haarolean and he said that if there are already other buttons in the white one, then we do it as a separate task, so this two buttons we can do with separate task Screen Shot 2023-03-01 at 15 41 15 Screen Shot 2023-03-01 at 15 41 29

will create a separate task for it

**Comment by armenuikafka** [comment]
@David-DB88 

- Connector "delete" should be white as in a design 
connector delete

@Haarolean after this fix we can merge I think
p.s. some design inconsistencies I would like to discuss with you and will report in a separate task

**Comment by David-DB88** [comment]
@armenuikafka i made a changes

**Review by armenuikafka — APPROVED** [review]

File: kafka-ui-react-app/src/components/PageContainer/PageContainer.tsx [hunk 8]
```diff
@@ -5,21 +5,23 @@ import * as S from 'components/PageContainer/PageContainer.styled';
...
+const PageContainer: React.FC<
+  PropsWithChildren<{ setDarkMode: (value: boolean) => void }>
+> = ({ children, setDarkMode }) => {
   const {
     value: isSidebarVisible,
     toggle,
     setFalse: closeSidebar,
   } = useBoolean(false);
-  const location = useLocation();
+  const location = useLocation()
```
Comment (Mgrdich) [inline kafka-ui-react-app/src/components/PageContainer/PageContainer.tsx:16]: please check this linter issue

File: kafka-ui-react-app/src/components/Topics/Topic/Messages/PreviewModal.styled.ts [hunk 9]
```diff
@@ -33,4 +33,5 @@ export const Field = styled.div`
   white-space: nowrap;
   overflow: hidden;
   margin-right: 5px;
+  color: ${({ theme }) => theme.modal.color}; ;
```
Comment (Mgrdich) [inline kafka-ui-react-app/src/components/Topics/Topic/Messages/PreviewModal.styled.ts:36]: here as well.

**Review by Mgrdich — COMMENTED** [review]
And also there is `'JSX'` is not defined ,it should be checked cause as far as i know , it is included React Global types with namespace without actually import , maybe the issue is with the linter settings.

File: kafka-ui-react-app/src/components/common/Switch/Switch.tsx [hunk 10]
```diff
@@ -2,22 +2,34 @@ import React from 'react';
 
 import * as S from './Switch.styled';
 
-interface SwitchProps {
+export interface SwitchProps {
   onChange(): void;
   checked: boolean;
   name: string;
+  checkedIcon?: JSX.Element;
+  unCheckedIcon?: JSX.Element;
```
Comment (workshur) [inline kafka-ui-react-app/src/components/common/Switch/Switch.tsx:10]:
```suggestion
  unCheckedIcon?: React.ReactNode;
```

File: kafka-ui-react-app/src/components/common/Switch/Switch.tsx [hunk 11]
```diff
@@ -2,22 +2,34 @@ import React from 'react';
 
 import * as S from './Switch.styled';
 
-interface SwitchProps {
+export interface SwitchProps {
   onChange(): void;
   checked: boolean;
   name: string;
+  checkedIcon?: JSX.Element;
```
Comment (workshur) [inline kafka-ui-react-app/src/components/common/Switch/Switch.tsx:9]:
```suggestion
  checkedIcon?: React.ReactNode;
```

File: kafka-ui-react-app/src/components/common/Select/Select.tsx [hunk 12]
```diff
@@ -17,10 +17,11 @@ export interface SelectProps {
   placeholder?: string;
   disabled?: boolean;
   onChange?: (option: string | number) => void;
+  isThemeMode?: boolean;
 }
 
 export interface SelectOption {
-  label: string | number;
+  label: string | number | JSX.Element;
```
Comment (workshur) [inline kafka-ui-react-app/src/components/common/Select/Select.tsx:24]:
```suggestion
  label: string | number | React.ReactNode;
```

**Review by workshur — COMMENTED** [review]
@David-DB88 

please run 

```
pnpm lint

pnpm test

pnpm tsc
```

**Comment by workshur** [comment]
Looks good so far

File: kafka-ui-react-app/src/components/common/Switch/Switch.tsx (hunk 11 above)
Comment (David-DB88) [reply kafka-ui-react-app/src/components/common/Switch/Switch.tsx]: ok

File: kafka-ui-react-app/src/components/common/Select/Select.tsx (hunk 12 above)
Comment (David-DB88) [reply kafka-ui-react-app/src/components/common/Select/Select.tsx]: ok

File: kafka-ui-react-app/src/components/PageContainer/PageContainer.tsx (hunk 8 above)
Comment (David-DB88) [reply kafka-ui-react-app/src/components/PageContainer/PageContainer.tsx]: checked

**Review by Haarolean — CHANGES_REQUESTED** [review]
image
still 61px

**Review by Haarolean — APPROVED** [review]
